magiclabs/create-magic-app
# CLI tool revamp

### 📦 Pull Request

This PR represents stage 1 of an overhaul of the `make-magic` command. 
* Removes outdated scaffolds
* Adds two new Nextjs templates that act as the starting point for Magic Auth and Magic Connect experiences
* Expands the templating functionality so that templates are chosen intelligently based on prompt answers rather than requiring users to explicitly select a template

We'll continue to roll out new templates and features moving forward. Plans include:
* Simplifying the process for external contributors to add their own templates, prompts, etc. 
* Removal of the `Zombi` dependency to reduce cognitive load for developers stepping in for the first time
* Improvements to analytics
* Addition of multiple frontend framework templates, web3 libraries, and more

This is only the beginning of a new and improved developer experience when starting new projects using Magic.

### 🚨 Test instructions

There are two ways to test the CLI app. You can run the staging environment using the following command in your command line:
```
npx @unboxed-software/make-magic-staging
```

Or you can run locally using the following steps:
1. Clone the source and open it in VSCode
2. Run `yarn` to install dependencies
3. Run `yarn build`
4. Choose a parent directory where you want a test project to be installed and navigate to it from the command line
5. Run `node [path/to/source]/bin.js`
6. The above should start the CLI tool. Follow the prompts until your template is cloned and ready to go
